docs/eBPFWeakReferences.md

```diff
@@ -0,0 +1,82 @@
+
+**Introduction and problem statement**
+
+The eBPF user-kernel interface is built around a few key objects such as 'Programs', 'Maps' and their interaction. The 'eBPF for Windows' implementation makes extensive use of 'reference counting' to manage the interactions and lifetimes of these objects.  Code paths 'acquire' references to the required objects for the needed duration and 'release' them when done.  The reference counting design ensures that the referred-to objects remain valid (i.e. are not deleted) for the duration of the reference acquisition.
```

Comment:
https://github.com/microsoft/ebpf-for-windows/blob/main/docs/DevelopmentGuide.md#formatting-notes says
> Source lines MUST NOT exceed 120 columns.

(One rationale is that github has problems showing later diffs in long lines)

Reply:
I'd assumed that was _code_ files only.  But now that you mention the GH problem, I can see why it's applicable to non-code files such as this one.  I'll fix this

Reply:
fixed